### ISSUE TITLE
Custom DNS authority support

### ISSUE THREAD

**Issue by elliots** [grpc/grpc-go#2173]
An error message was added in #2067 when the dns server is overridden according to https://github.com/grpc/grpc/blob/master/doc/naming.md

For go 1.8+ I have a custom resolver to use

```golang
&net.Resolver{
		PreferGo: true,
		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
			dialer := net.Dialer{}
			return dialer.DialContext(ctx, "udp", authority)
		},
}
```

while still erroring on 1.7 and below.

Should I submit a pull request? (or am I misunderstanding how the dns resolver works?)

Please answer these questions before submitting your issue.

### What version of gRPC are you using?

HEAD

### What version of Go are you using (`go version`)?

1.10

### What operating system (Linux, Windows, …) and version?

OSX

### What did you do?
If possible, provide a recipe for reproducing the error.

Used a custom dns authority

### What did you expect to see?

It resolves using my dns server

### What did you see instead?

It gives an error saying it is not supported

**Comment by dfawley** [comment]
@elliots The default DNS resolver does not support an authority, so it should error if one is set.  If I understand correctly, you should either always install your custom resolver or set the authority only when it is installed.

Note that https://github.com/grpc/grpc/pull/15618 is currently updating this documentation to explicitly call out authority as unsupported for DNS with the exception of the c-ares DNS resolver.

**Comment by elliots** [comment]
I meant, should I provide a pull-request that adds authority support to the default dns resolver? (at least, go 1.8+)

**Comment by dfawley** [comment]
@elliots - Yes, thanks!  If you send us a PR, we can review it and take in this functionality.

**Comment by dfawley** [comment]
Fixed by #2265.
